packages/quantic/force-app/main/default/lwc/quanticDocumentSuggestion/__tests__/quanticDocumentSuggestion.test.js

```diff
@@ -0,0 +1,371 @@
+jest.mock('c/quanticHeadlessLoader');
+
+import QuanticDocumentSuggestion from 'c/quanticDocumentSuggestion';
+import {buildCreateTestComponent, cleanup, flushPromises} from 'c/testUtils';
+import * as quanticHeadlessLoader from 'c/quanticHeadlessLoader';
+const headlessLoaderMock = jest.mocked(quanticHeadlessLoader);
+
+const engineMock = {
+  id: 'exampleEngineIdMock',
+  dispatch: jest.fn(),
+};
+headlessLoaderMock.initializeWithHeadless.mockImplementation(
+  async (element, _, initialize) => {
+    initialize(engineMock);
+  }
+);
+headlessLoaderMock.getHeadlessEnginePromise.mockImplementation(() =>
+  Promise.reject({message: 'Skip initialization'})
+);
+
+let updateSuggestions;
+const documentSuggestionListMock = {
+  subscribe: jest.fn((callback) => {
+    updateSuggestions = async () => {
+      callback();
+      await flushPromises();
+    };
+    return () => {}; // Fake unsubscribe function
+  }),
+};
+
+const functionsMocks = {
+  fetchDocumentSuggestions: jest.fn(),
+  logDocumentSuggestionClick: jest.fn(),
+  logDocumentSuggestionRating: jest.fn(),
+};
+
+const headlessCaseAssistMock = {
+  buildDocumentSuggestionList: jest
+    .fn()
+    .mockReturnValue(documentSuggestionListMock),
+  loadCaseAssistAnalyticsActions: jest.fn().mockReturnValue({}),
+  loadDocumentSuggestionActions: jest.fn().mockReturnValue({
+    fetchDocumentSuggestions: functionsMocks.fetchDocumentSuggestions,
+    logDocumentSuggestionClick: functionsMocks.logDocumentSuggestionClick,
+    logDocumentSuggestionRating: functionsMocks.logDocumentSuggestionRating,
```

Comment:
To stay lawful to the Headless implementation, logDocumentSuggestionClick and logDocumentSuggestionRating should come from  `loadCaseAssistAnalyticsActions ` and not `loadDocumentSuggestionActions `: https://github.com/coveo/ui-kit/blob/77feadef7bd45d77713ef65d761bc5161c77f2ab/packages/headless/src/features/case-assist/case-assist-analytics-actions-loader.ts#L116-L130